### ISSUE TITLE
`type-annotation-spacing`: recursive conflict with `key-spacing`

### ISSUE THREAD

**Issue by ngdangtu-vn** [eslint-stylistic/eslint-stylistic#1219]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I have this snippet:

```ts
// component.tsx
export interface CharacterProps {
   data: Array<{
      name  : string
      age   : string
      gender: string
      race  : string
   }>
}
```

If I enable the 2 rules `@stylistic/key-spacing: ['error', { 'align': 'colon' }]` and `@stylistic/type-annotation-spacing: ['error', { 'after': true }]`, it will break the eslint:

```log
2026-05-29 69:69:69.404 [error] (node:3269630) ESLintCircularFixesWarning: Circular fixes detected while fixing /home/user/repo/component.tsx. It is likely that you have conflicting rules in your configuration.
```

So I have to disable `@stylistic/type-annotation-spacing` in order to favour `@stylistic/key-spacing`. If I do that then it fails to format this code:

```ts
function split_tx(raw:string, start   : number) { }
```

### Reproduction

https://github.com/ngdangtu-vn/eslint-warehouse/tree/style/reprod/conflict---type-annotation-spacing---key-spacing

### System Info

```shell
System:
    OS: Linux 6.17 TUXEDO OS 24.04.4 LTS
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700H
    Memory: 2.50 GB / 31.09 GB
    Container: Yes
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 22.13.0 - /home/user/.local/opt/nvm/versions/node/v22.13.0/bin/node
    Yarn: 1.22.22 - /home/user/.local/opt/pnpm/yarn
    npm: 10.9.2 - /home/user/.local/opt/nvm/versions/node/v22.13.0/bin/npm
    pnpm: 10.24.0 - /home/user/.local/opt/pnpm/pnpm
    bun: 1.3.13 - /home/user/.local/bin/bun
    Deno: 2.5.6 - /home/user/.local/bin/deno
  npmPackages:
    @stylistic/eslint-plugin: ^5.10.0 => 5.10.0 
    eslint: ^10.4.0 => 10.4.0
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by ngdangtu-vn** [comment]
Do you guys know where is the logic for this rule? I'm a bit curious about it.

**Comment by 9romise** [comment]
The source code for this rule is at https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/type-annotation-spacing/type-annotation-spacing.ts. I suggest adding an `'ignore'` option to `spacingConfig` to allow ignoring.

**Comment by ngdangtu-vn** [comment]
That seems like a good idea. Do we have any similar case like this before? I want to refer it for a naming and design consistency.

---

After searching through the doc a bit, I think the option `multiline` could solve the problem and keep the consistency for the whole design. What do you think @9romise?

**Comment by ngdangtu-vn** [comment]
So I double check the doc again and here is what I thought:


1. `overrides` option doesn't turn off the feature and still cause bug

```js
{
   'overrides': { 'property': { 'before': false, 'after': false } }
}
```

2. `multiline` sounds like a quick solution but it doesn't feel very scalable and fit to the rule option

```js
{
   rules : {
      '@stylistic/type-annotation-spacing': ['error', { 'multiline': false }],
   }
}
```

3. 

```ts
interface TypeAnnotationSpacingSchema0 {
  before?: boolean
  after?: boolean
  overrides?: {
    colon?: SpacingConfig
    variable?: SpacingConfig
    parameter?: SpacingConfig
    property?: SpacingConfig
    returnType?: SpacingConfig
    questionMark?: SpacingConfig
  }
  disable?: {
    colon?: boolean
    variable?: boolean
    parameter?: boolean
    property?: boolean // ← default is true
    returnType?: boolean
    questionMark?: boolean
  }
}
```

I think we can safely set `disable.property` as `false` by default and the rest is `true` right?

**Comment by 9romise** [comment]
```
interface TypeAnnotationSpacingSchema0 {
  before?: boolean
  after?: boolean
  overrides?: {
    colon?: SpacingConfig
    variable?: SpacingConfig
    parameter?: SpacingConfig
    property?: SpacingConfig
    returnType?: SpacingConfig
    questionMark?: SpacingConfig
  }
}
```

I'd prefer extends the `SpacingConfig` here as

```
{ before?: boolean, after?: boolean } | 'ignore'
```

**Comment by ngdangtu-vn** [comment]
I got your point now. That could work with minimum update on the end users. Lemme see what I can do with this. Thanks for suggestion :D
